controllers/progressiverollout_controller.go

```diff
@@ -92,11 +94,21 @@ func (r *ProgressiveRolloutReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 
 		// Get the Applications to update
-		scheduledApps := scheduler.Scheduler(apps, stage)
+		scheduledApps := scheduler.Scheduler(outOfSyncApps, stage)
 
 		for _, s := range scheduledApps {
-			// TODO: add sync method here
 			r.Log.Info("syncing app", "app", s)
+
+			_, err := r.syncApp(s.Name)
+
+			if err != nil {
+				if strings.Contains(err.Error(), "another operation is already in progress") {
+					log.Info("another operation is already in progress", "app", s)
```

Comment:
I assume this means we proceed as normal - it's not an error _per se_ - so I don't return here.

Reply:
I'm not really even sure we want to log this to be honest. This mean we tried to re sync the app while a sync was already in progress, correct? So what's the big deal? :)
I might miss a valid use case.